packages/alfa-dom/src/node/document.ts

```diff
@@ -38,22 +41,13 @@ export class Document extends Node<"document"> {
     return this._frame;
   }
 
-  private _elementDescendants: Array<Sequence<Element>> = [];
   /**
    * {@link https://dom.spec.whatwg.org/#concept-tree-descendant}
    */
-  // We very often need all elements in a document, typically at the start of
-  // a rule Applicability. Caching the filtering saves a lot of time.
   public elementDescendants(
     options: Node.Traversal = Node.Traversal.empty
   ): Sequence<Element> {
-    if (this._elementDescendants[options.value] === undefined) {
-      this._elementDescendants[options.value] = this.descendants(
-        options
-      ).filter(Element.isElement);
-    }
-
-    return this._elementDescendants[options.value];
+    return getElementDescendants(this, options);
```

Comment:
**note:** Not sure if we should replace all usages of `elementDescendants` with `getElementDescendant` instead of having this indirection. If so, maybe that can be done in a separate PR to limit the blast radius of this one. OTOH it's also convenient to have this instance function so that we can do `foo.elementDescendants()` 🤔

Reply:
I think hat ultimately we should. `foo.elementDescendants` wasn't that well thought through.
Let's keep it for another PR indeed.

Reply:
Cool, I'll add it to #1037